### PR TITLE
Suppress EntryPointNotFoundException on sentry native

### PR DESCRIPTION
**Pull request by bruno-garcia**
Resolves: https://github.com/getsentry/sentry-unity/issues/1864

### CODE REVIEW

File: package-dev/Runtime/SentryInitialization.cs [hunk 1]
```diff
@@ -123,10 +123,18 @@ public static void Init()
...
+            {
+                // At this point Unity has taken the signal handler and will not invoke our handler. So we register our
+                // backend once more to make sure user-defined data is available in the crash report and the SDK is able
+                // to capture the crash.
+                SentryNative.ReinstallBackend();
+            }
+            catch (EntryPointNotFoundException e)
+            {
+                // See: https://github.com/getsentry/sentry-unity/issues/1864
+                // TODO: if option.Debug: true, debug log this
```
Comment (bruno-garcia) [inline package-dev/Runtime/SentryInitialization.cs:136]: @bitsandfoxes is there a static logger I can try to grab to do this?

File: src/Sentry.Unity.Native/SentryNative.cs [hunk 2]
```diff
@@ -80,7 +83,26 @@ public static void Configure(SentryUnityOptions options, ISentryUnityInfo sentry
...
+        try
+        {
+            // At this point Unity has taken the signal handler and will not invoke our handler. So we register our
+            // backend once more to make sure user-defined data is available in the crash report and the SDK is able
+            // to capture the crash.
+            ReinstallBackendFunction?.Invoke();
+        }
+        catch (EntryPointNotFoundException e)
+        {
+            SentrySdk.CaptureException(new Exception("The SDK failed to reinstall the backend and won't capture native errors.", e));
```
Comment (bitsandfoxes) [inline src/Sentry.Unity.Native/SentryNative.cs:105]: I'm not a huge fan of capturing exceptions the SDK throws itself but this is during initialization, making sure that the lack of observability of native crashes is known.

Comment (bruno-garcia) [reply src/Sentry.Unity.Native/SentryNative.cs:105]: I'm not sure this is the best route. What would the user do with this exception?
Also, they'd be paying for these events.

Debug log is the way to go imo

File: src/Sentry.Unity.Native/SentryNative.cs [hunk 3]
```diff
@@ -80,7 +83,26 @@ public static void Configure(SentryUnityOptions options, ISentryUnityInfo sentry
             }
         }
         options.CrashedLastRun = () => crashedLastRun;
+
+        // The backend will only be reinstalled if the native SDK has been initialized successfully.
+        // We're handing this off to be called in `BeforeSceneLoad` instead of `SubsystemRegistration` as it's too soon
+        // and the SignalHandler would still get overwritten.
+        ReinstallBackendFunction = SentryNativeBridge.ReinstallBackend;
```
Comment (bruno-garcia) [inline src/Sentry.Unity.Native/SentryNative.cs:90]: This seems hacky: do we change the method in any way? We're just signaling from this part of the case to another. I think a bool is more clear and has other benefits (as in not having the action)

File: src/Sentry.Unity.Native/SentryNative.cs (hunk 2 above)
Comment (bitsandfoxes) [reply src/Sentry.Unity.Native/SentryNative.cs:105]: How would logging help? I think we have to make sure users can be certain the SDK is functioning correctly.
The scenario I've got in my head and would like to cover:
1. User ships their game but the native SDK fails to reinstall.
2. Crash free rate is 100%
3. Players still report crashes but the user has no idea why Sentry is not reporting any
4. What now? Enable `debug` and ask for logs?

We're doing the same thing to cover the native SDK's `Init` here
https://github.com/getsentry/sentry-unity/blob/8c0967fedc553a2bd0f0ab95ece8b21b6c47e9d2/package-dev/Runtime/SentryInitialization.cs#L77-L92

File: src/Sentry.Unity.Native/SentryNative.cs [hunk 4]
```diff
@@ -13,6 +14,8 @@ public static class SentryNative
 {
     private static readonly Dictionary<string, bool> PerDirectoryCrashInfo = new();
 
+    private static bool ShouldReinstallBackend;
```
Comment (bruno-garcia) [inline src/Sentry.Unity.Native/SentryNative.cs:17]:
```suggestion
    private static bool _shouldReinstallBackend;
```

File: src/Sentry.Unity.Native/SentryNative.cs [hunk 5]
```diff
@@ -80,7 +83,31 @@ public static void Configure(SentryUnityOptions options, ISentryUnityInfo sentry
             }
         }
         options.CrashedLastRun = () => crashedLastRun;
+
+        ShouldReinstallBackend = true;
```
Comment (bruno-garcia) [inline src/Sentry.Unity.Native/SentryNative.cs:87]:
```suggestion
        _shouldReinstallBackend = true;
```

File: src/Sentry.Unity.Native/SentryNative.cs [hunk 6]
```diff
@@ -80,7 +83,31 @@ public static void Configure(SentryUnityOptions options, ISentryUnityInfo sentry
...
     }
 
-    public static void ReinstallBackend() => SentryNativeBridge.ReinstallBackend();
+    // We're calling this in `BeforeSceneLoad` instead of `SubsystemRegistration` as it's too soon and the
+    // SignalHandler would still get overwritten.
+    [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.BeforeSceneLoad)]
+    private static void ReinstallBackend()
+    {
+        // The backend should only be reinstalled if the native SDK has been initialized successfully.
+        if (!ShouldReinstallBackend)
```
Comment (bruno-garcia) [inline src/Sentry.Unity.Native/SentryNative.cs:96]:
```suggestion
        if (!_shouldReinstallBackend)
```

**Pull request closed without merging**